src/base/net/downloadhandler.cpp

```diff
@@ -92,8 +92,10 @@ void DownloadHandler::processFinishedDownload()
             // Success
             QByteArray replyData = m_reply->readAll();
             if (m_reply->rawHeader("Content-Encoding") == "gzip") {
-                // uncompress gzip reply
-                Utils::Gzip::uncompress(replyData, replyData);
+                // decompress gzip reply
+                QByteArray tmp;
```

Comment:
Also I plan to add a compression level variable to compress:
`QByteArray compress(const QByteArray &data, int level = 6, bool *ok = nullptr)`.

As you can see from https://github.com/inikep/lzbench#benchmarks , `level=9` is really slow even on modern CPU. I would trade much faster compression for less worse compressed ratio.